csharp/src/Apache.Arrow/ArrowBuffer.BitmapBuilder.cs

```diff
@@ -89,6 +89,37 @@ public BitmapBuilder Append(bool value)
                 return this;
             }
 
+            /// <summary>
+            /// Append a span of bits.
+            /// </summary>
+            /// <param name="source">Source of bit span.</param>
+            /// <param name="validBits">Number of valid bits in the bit span</param>
+            /// <returns>Returns the builder (for fluent-style composition).</returns>
+            public BitmapBuilder Append(ReadOnlySpan<byte> source, int validBits)
+            {
+                var length = source.IsEmpty ? validBits : Math.Min(source.Length * 8, validBits);
+
+                // Check if memory copy can be used from the source array (performance optimization for byte-aligned coping)
+                if (!source.IsEmpty && this.Length % 8 == 0)
+                {
+                    EnsureAdditionalCapacity(length * 8);
+                    source.Slice(0, BitUtility.ByteCount(length)).CopyTo(Span.Slice(Length / 8));
+                    
+                    Length += length;
+                    SetBitCount = BitUtility.CountBits(Span, 0, Length);
```

Comment:
Do we really need to count all the bits again? Can't we just count the bits that were appended and add it to the existing number?

Reply:
Agree. Fixed